### PR TITLE
Crm457 626 remove all uplifts

### PR DESCRIPTION
**Pull request by ivanELEC**
## Description of change

- Add form for remove all work items uplifts
- Add link to button on work items to for remove all uplifts
- Add conditional logic to show/not show button depending on whether any work items with uplifts from providers exists on the claim
- Add work_items/uplift controller to handle form
- Add validations and error messages to form
- Fix Letters and Calls and Work Items remove all uplifts form error messaging for when explanation is blank

## Link to relevant ticket

[Remove Uplifts](https://dsdmoj.atlassian.net/browse/CRM457-626)
[Remove Uplifts Event Stream](https://dsdmoj.atlassian.net/browse/CRM457-627)

## Notes for reviewer
I've moved the grouping of the work items to the view because the work items base array is needed to find out of there are uplifts or not - let me know if there's a preferred way to factor this

### CODE REVIEW

**Review by imtiazAhmd — DISMISSED** [review]

File: app/views/work_items/index.html.erb [hunk 1]
```diff
@@ -20,11 +20,11 @@
...
   <% end %>
 
-  <% if claim.editable? %>
-    <%= button_to(t(".remove_uplift_all"), '#', class: "govuk-button govuk-button--secondary", data_module: "govuk-button") %>
+  <% if claim.editable? && work_items.any? { |work_item| work_item.uplift? } %>
+    <%= govuk_button_link_to(t(".remove_uplift_all"), edit_claim_work_items_uplift_path(claim), { secondary: true, data: { turbo: 'false' } }) %>
   <% end %>
 
-  <% work_items.each do |completed_on, work_items_for_date| %>
+  <% work_items.group_by { |work_item| work_item[:completed_on] }.each do |completed_on, work_items_for_date| %>
```
Comment (dwhenry) [inline app/views/work_items/index.html.erb:27]:
```
<% work_items.group_by(&:completed_on).each do |completed_on, work_items_for_date| %>
```
should be the same

File: db/seeds/de2d6169-c769-4571-b3f2-220bcf8c3ce9/version.json [hunk 2]
```diff
@@ -59,7 +59,7 @@
     "work_items": [
       {
         "id": "cf5e303e-98dd-4b0f-97ea-3560c4c5f137",
-        "uplift": 10,
+        "uplift": 0,
```
Comment (dwhenry) [inline db/seeds/de2d6169-c769-4571-b3f2-220bcf8c3ce9/version.json:62]: any reason for this?

**Review by dwhenry — APPROVED** [review]
nothing major - looks good.. :)

File: db/seeds/de2d6169-c769-4571-b3f2-220bcf8c3ce9/version.json (hunk 2 above)
Comment (ivanELEC) [reply db/seeds/de2d6169-c769-4571-b3f2-220bcf8c3ce9/version.json:62]: just wanted to be able to test work item uplifts tab where there were no uplifts (therefore no remove all uplifts button)

File: app/views/work_items/index.html.erb (hunk 1 above)
Comment (ivanELEC) [reply app/views/work_items/index.html.erb:27]: will fix

**Review by imtiazAhmd — APPROVED** [review]

**Pull request merged**